### PR TITLE
fix(storage): blob metadata to None regression

### LINKED ISSUE #10107: Storage: Possible metadata regression on blobs in 1.24.0
#### Environment details

* Debian 10
* google-cloud-storage version: 1.24.1

#### Steps to reproduce

It's documented that metadata that isn't set will return a NoneType. Naturally, when wanting to unset any metadata, you expect to be able to pass in `None`. This worked up until 1.24.0 which broke this with #9796.

I think you can still technically set the metadata to an empty dictionary and get the same functionality, but it's a bit counter intuitive, when no metadata is described as being a `NoneType`.

#### Code example

```python
storage_client = storage.Client(...)
blob = storage_client.get_bucket('abc').get_blob('abc')
blob.metadata = None
blob.patch()
```

#### Stack trace

```
AttributeError: 'NoneType' object has no attribute 'items'
  File "django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "api/media/views.py", line 96, in put
    product.media.set_primary(name)
  File "media/api.py", line 134, in set_primary
    blob.metadata = None
  File "/usr/local/lib/python3.7/site-packages/google/cloud/storage/blob.py", line 1917, in metadata
    value = {k: str(v) for k, v in value.items()}
```

### PR DESCRIPTION
**Pull request by IlyaFaer**
Fixes #10107

### CODE REVIEW

**Comment by IlyaFaer** [comment]
Test_Blob.test_get_iam_policy failed - not related to this PR

File: storage/google/cloud/storage/blob.py [hunk 1]
```diff
@@ -1910,7 +1910,8 @@ def metadata(self, value):
         :type value: dict
         :param value: (Optional) The blob metadata to set.
         """
-        value = {k: str(v) for k, v in value.items()}
+        if value is not None:
+            value = {k: str(v) for k, v in value.items()}
```
Comment (IlyaFaer) [inline storage/google/cloud/storage/blob.py]: `value` has tag `Optional`, so it looks like regression
Do we need to do the same thing for bucket labels? I don't see `Optional` tag for them

Comment (frankyn) [reply storage/google/cloud/storage/blob.py]: I think the documentation is incorrect. The code is correct in that the property metadata is optional but should not be optional when setting this value.

Comment (IlyaFaer) [reply storage/google/cloud/storage/blob.py]: Okay, deleted `Optional` tag. As I see, it was added by some kind of a script

**Pull request closed without merging**